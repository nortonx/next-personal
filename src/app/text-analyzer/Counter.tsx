interface CounterProps {
  label: string;
  counter: number;
}
export function Counter({ label, counter } : CounterProps){
    return(
        <div data-testid="counter">
          <h1>{label}</h1>
          <h2>{counter}</h2>
        </div>
    )
}