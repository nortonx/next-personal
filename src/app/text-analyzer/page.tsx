'use client'

import { useState } from "react";
import { Container, Row } from "react-bootstrap";
import { Counter } from "./Counter";
import { Textarea } from "@/components/Textarea";


const TextAnalyzer = () => {
    const [content, setContent] = useState("")

    function handleOnChangeText(content: string) {
        setContent(content)
    }

    function wordCounter(content: string) {
        if (content) {
            const words = content.trim().split(" ")
            return words?.length
        }
        return 0
    }

    function characterCounter(content: string) {
        const characters = content.split("")
        return characters?.length ?? 0
    }

    function sentencesCounter(content: string) {
        const sentences = content.match(/[^\.!\?]+[\.!\?]+/g);
        return sentences?.length ?? 0
    }

    function paragraphsCounter(content: string) {
        if (content) {
            const paragraphs = content.split(/\r?\n/)
            return paragraphs?.length
        }
        return 0
    }

    return(
        <main data-testid="text-analyzer">
            <Container fluid data-testid="analyzer">
                <Row xs={1} md={4}>
                    <Counter 
                        label="Words"
                        counter={wordCounter(content)}
                    />
                    <Counter
                        label="Characters"
                        counter={characterCounter(content)}
                    />
                    <Counter
                        label="Sentences"
                        counter={sentencesCounter(content)}
                    />
                    <Counter
                        label="Paragraphs"
                        counter={paragraphsCounter(content)}
                    />
                </Row>
                <Row>
                    <Textarea 
                      label="Type in your text"
                      id="text-analyzer"
                      onChangeText={handleOnChangeText}
                    />
                </Row>
            </Container>
        </main>
    )
}

export default TextAnalyzer;