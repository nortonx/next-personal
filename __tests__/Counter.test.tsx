import { render, screen } from '@testing-library/react'
import { Counter } from '@/app/text-analyzer/Counter'

describe("Counter component", () => {
  it("Should render component and match snapshot", async () => {
    render(<Counter label="Words" counter={0}/>)
    const element = await screen.getByTestId('counter');
    expect(element).toBeInTheDocument()
    expect(element).toMatchSnapshot();
  })
})