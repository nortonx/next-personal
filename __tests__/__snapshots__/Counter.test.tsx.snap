// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Should render component and match snapshot 1`] = `
<div
  class="py-2"
  data-testid="counter"
>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="d-flex justify-content-center fs-2 card-title h5"
      >
        Words
      </div>
      <p
        class="d-flex justify-content-center text-secondary fs-3 card-text"
      >
        0
      </p>
    </div>
  </div>
</div>
`;
